### PR TITLE
Fix relationship dropdown error & encoding

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes: #5740 (confirmed by 2 users)
Fixes: https://wordpress.org/support/topic/warning-ltrim-expects-parameter-1-to-be-string/

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: classes/fields/file.php [hunk 1]
```diff
@@ -562,7 +562,7 @@ public function build_dfv_field_item_data( $args ) {
 			$data[] = array(
 				'id'        => esc_html( $id ),
 				'icon'      => esc_attr( $icon ),
-				'name'      => esc_html( $title ),
+				'name'      => html_entity_decode( esc_html( $title ) ),
```
Comment (sc0ttkclark) [inline classes/fields/file.php:565]: I’m not sure this is the right way, it seems to undo the work esc_html does

File: classes/fields/pick.php [hunk 2]
```diff
@@ -1381,12 +1386,12 @@ public function build_dfv_field_item_data_recurse_item( $item_id, $item_title, $
...
 		$item = array(
-			'id'           => esc_html( $item_id ),
-			'icon'         => esc_attr( $icon ),
-			'name'         => esc_html( $item_title ),
-			'edit_link'    => esc_url( $edit_link ),
-			'link'         => esc_url( $link ),
-			'selected'     => $selected,
+			'id'        => esc_html( $item_id ),
+			'icon'      => esc_attr( $icon ),
+			'name'      => html_entity_decode( esc_html( $item_title ) ),
```
Comment (sc0ttkclark) [inline classes/fields/pick.php:1391]: I’m not sure this is the right way, it seems to undo the work esc_html does

File: classes/fields/file.php (hunk 1 above)
Comment (sc0ttkclark) [reply classes/fields/file.php:565]:
```suggestion
				'name'      => esc_html( wp_kses_post( html_entity_decode( $title ) ) ),
```

File: classes/fields/pick.php (hunk 2 above)
Comment (sc0ttkclark) [reply classes/fields/pick.php:1391]:
```suggestion
			'name'      => esc_html( wp_kses_post( html_entity_decode( $title ) ) ),
```

**Comment by sc0ttkclark** [comment]
Merged into `2.x` and reset `master`

**Comment by aschulze1969** [comment]
Hello,
I had the same issues with showing &nbsp; instead of an tab/indent as described in the post #5753 
After update to 2.7.21 the dropdown with the relationship shows nothing for hierachical items. Only the items of the first level will be shown. All lines with the second, third.. level are completly empty.

**Pull request merged**